### PR TITLE
 ops: tpetra: revise rank-1 update

### PR DESCRIPTION
**Pull request by mzuzek**
Contents:
- [x] revised overload constraints 
- [x] added extent check
- [x] added explicit type casting for coefficients
- [x] added NaN injection unit tests (impl is fine - no fix needed)
- [x] removed duplicated fixtures in unit tests

refs #448

### CODE REVIEW

File: tests/functional_small/ops/ops_tpetra_vector.cc [hunk 1]
```diff
@@ -386,62 +333,112 @@ TEST_F(tpetraVectorGlobSize15Fixture, vector_update4_b)
...
-TEST(ops_tpetra, vector_elementwiseMultiply)
+TEST_F(ops_tpetra, vector_update_nan1)
+{
+    auto v = pressio::ops::clone(*myVector_);
+    auto v2_h = v.getLocalViewHost(Tpetra::Access::ReadOnly);
+    auto v_h = Kokkos::subview(v2_h, Kokkos::ALL, 0);
+    pressio::ops::fill(v, 1.);
+    auto a = pressio::ops::clone(*myVector_);
+    pressio::ops::fill(a, 1.);
+    auto nan = pressio::ops::clone(*myVector_);
```
Comment (fnrizzi) [inline tests/functional_small/ops/ops_tpetra_vector.cc:344]: can you please rename: `nan -> vecOfNan`

**Review by fnrizzi — CHANGES_REQUESTED** [review]
see comment

File: tests/functional_small/ops/ops_tpetra_vector.cc (hunk 1 above)
Comment (mzuzek) [reply tests/functional_small/ops/ops_tpetra_vector.cc]: ok, fixed on 5bcb6c5b0b9e618cc4bf51aa0cdcd7b0b314b274

**Review by fnrizzi — APPROVED** [review]

**Comment by mzuzek** [comment]
thanks @mzuzek for your review and comments!

**Pull request merged**